src/cast_core/src/MonotonicLoader.cpp

```diff
@@ -109,27 +109,42 @@ void genny::actor::MonotonicLoader::run() {
                     }
                     totalOpCtx.success();
                 }
-                // For each index
+                // Make the index
+                bool _indexReq = false;
+                builder::stream::document builder{};
+                auto in_array = builder << "createIndexes" << collectionName
+                                        <<  "indexes" << builder::stream::open_array;
                 for (auto&& [keys, options] : config->indexes) {
-                    // Make the index
+                    _indexReq = true;
                     auto indexKey = keys();
-                    BOOST_LOG_TRIVIAL(debug)
-                        << "Building index " << bsoncxx::to_json(indexKey.view());
                     if (options) {
                         auto indexOptions = (*options)();
-                        BOOST_LOG_TRIVIAL(debug)
-                            << "With options " << bsoncxx::to_json(indexOptions.view());
-                        auto indexOpCtx = _indexBuild.start();
-                        collection.create_index(std::move(indexKey), std::move(indexOptions));
-                        indexOpCtx.success();
+                        in_array = in_array << builder::stream::open_document
+                                            << "key" << indexKey.view()
+                                            << builder::concatenate(indexOptions.view())
+                                            << builder::stream::close_document;
                     } else {
-                        auto indexOpCtx = _indexBuild.start();
-                        collection.create_index(std::move(indexKey));
-                        indexOpCtx.success();
+                        std::string index_name = "";
+                        for (auto field : indexKey.view()) {
+                            index_name = index_name + field.key().to_string();
+                        }
+                        in_array = in_array << builder::stream::open_document
+                                            << "key" << indexKey.view()
+                                            << "name" << index_name
+                                            << builder::stream::close_document;
                     }
                 }
+                auto after_array = in_array << builder::stream::close_array;
+                document::value doc = after_array << builder::stream::finalize;
+                if (_indexReq) {
+                    BOOST_LOG_TRIVIAL(debug)
+                            << "Building index" << to_json(doc.view());
+                    auto indexOpCtx = _indexBuild.start();
+                    config->database.run_command(doc.view());
+                    indexOpCtx.success();
```

Comment:
Also a note that this will change the nature of this recording, since this op is now tracking a single parallelized index creation rather than many separate ones. Might be worth mentioning in #performance-tooling-users so that people know / can object if needed.

Reply:
Yes will do that once I merge the changes